File: ifw-api/src/main/java/com/merxury/ifw/IntentFirewall.java
package com.merxury.ifw;

import com.merxury.ifw.entity.ComponentType;

public interface IntentFirewall {
    void save() throws Exception;

    boolean add(String packageName, String componentName, ComponentType type);

    boolean remove(String packageName, String componentName, ComponentType type);

    boolean getComponentEnableState(String packageName, String componentName);

    void clear();

    void clear(String name);

    void reload();
}

File: ifw-api/src/main/java/com/merxury/ifw/IntentFirewall.kt
package com.merxury.ifw

import com.merxury.ifw.entity.ComponentType

interface IntentFirewall {
    @Throws(Exception::class)
    suspend fun save()

    @Throws(Exception::class)
    suspend fun load(): IntentFirewall
    suspend fun add(packageName: String, componentName: String, type: ComponentType?): Boolean
    suspend fun remove(packageName: String, componentName: String, type: ComponentType?): Boolean

    /**
     * @return false if the component is blocked
     */
    suspend fun getComponentEnableState(packageName: String, componentName: String): Boolean

    @Throws(Exception::class)
    suspend fun clear()
    val packageName: String
}